lib/http3/common.c

```diff
@@ -57,6 +57,51 @@ struct st_h2o_http3_ingress_unistream_t {
 
 const ptls_iovec_t h2o_http3_alpn[1] = {{(void *)H2O_STRLIT("h3-25")}};
 
+static struct {
+    pthread_mutex_t lock;
+    time_t last_reported;
+    size_t failures;
+    size_t successes;
+    int last_errno;
+} track_sendmsg = {
+    .lock = PTHREAD_MUTEX_INITIALIZER,
+};
+
+static void track_sendmsg_failure(void)
+{
+    __sync_fetch_and_add(&track_sendmsg.failures, 1);
+    track_sendmsg.last_errno = errno;
+}
+
+static void track_sendmsg_success(void)
+{
+    __sync_fetch_and_add(&track_sendmsg.successes, 1);
+}
+
+static void track_sendmsg_flush(h2o_loop_t *loop)
+{
+    uint64_t now = h2o_now(loop);
+    if (track_sendmsg.last_reported + 60000 < now) {
+        /* this is racy, and might under report successes when there is a
+           failure, but allows to avoid taking the lock in the happy path */
+        if (track_sendmsg.failures == 0) {
+            track_sendmsg.last_reported = now;
+            track_sendmsg.successes = 0;
```

Comment:
I think it might be a good idea to define `successes` and `failures` as counters from the start of the process, rather than counters from the previous log emission. That avoids the problem of resetting the values.

When it becomes time to log, we can simply copy the values, compare them with the previous copy, and log the difference.